### DISCUSSION TITLE
Interest in more builtin HTTP extractors

### DISCUSSION THREAD

**Discussion by jordanisaacs** [tokio-rs/axum#366] [Ideas]
I am working on building a tracing library for Axum and one thing that I noticed coming from Actix Web is that many ease of use extractors are missing for requests. For example Actix has an implementation to match the pattern of the request, and a `ConnectionInfo` which parses the headers for you. I feel this would be best suited to be part of Axum but I am curious about other people's thoughts.

For reference here is Actix's HttpRequest documentation: https://docs.rs/actix-web/3.3.2/actix_web/struct.HttpRequest.html#method.connection_info

**Comment by davidpdrsn** [comment]
What specific things are you looking for?

Seems to be axum has the things in `actix_web::dev::ConnectionInfo` already:

- [`Router::into_make_service_with_connect_info`](https://docs.rs/axum/0.2.6/axum/routing/struct.Router.html#method.into_make_service_with_connect_info) gives you remote address
- [`Uri`](https://docs.rs/http/0.2.5/http/uri/struct.Uri.html) gives you the URI including the scheme
- [`HeaderMap`](https://docs.rs/http/0.2.5/http/header/struct.HeaderMap.html) and [`TypedHeader`](https://docs.rs/axum/0.2.6/axum/extract/struct.TypedHeader.html) gives you headers
